change asdf.tests.helpers uses to asdf.testing.helpers

braingram opened the pull request:
asdf.tests.helpers will be deprecated:
https://github.com/asdf-format/asdf/pull/1440

The uses in specutils are limited to assert_roundtrip_tree

A similar roundtrip_object exists in asdf.testing.helpers which serializes then deserializes an object (without asserting equality). This commit replaces the uses of assert_roundtrip_tree with roundtrip_object and uses the equality comparison built into the custom asdf types implemented in this module to check for equality. These equality comparisons are the same checks that were used internally by assert_roundtrip_tree.

pllim commented on specutils/io/asdf/tags/tests/test_spectra.py:
```diff
@@ -26,27 +27,26 @@ def create_spectrum1d(xmin, xmax, uncertainty=None):
-    assert_roundtrip_tree(tree, tmpdir)
+    # there is no implemented asdf type for SpectralAxis and no defined
+    # equality comparison (assert_equal)
+    assert roundtrip_object(spectral_axis) == spectrum_axis
```
Will this actually pass now? This test is marked as xfail.

pllim replied:
Nope, we have xfail strict on and CI passed, so this is still xfail

https://github.com/astropy/specutils/blob/ca3153f05b8d3613bba0b865e273187e8ff176c4/setup.cfg#L52

pllim commented:
Can u pls fix code style? Thx!

braingram commented:
> Can u pls fix code style? Thx!

I believe I fixed the code style issues. It pointed out a typo in the xfailed test (I used `spectrum_axis` instead of `spectral_axis`). I fixed that typo and the test still fails because the equality comparison fails.

Based on https://github.com/astropy/specutils/pull/645#issuecomment-614271632 SpectralAxis fails to round_trip and returns as SpectralCoord. I changed the test to fail due to a type comparison and added a link to the comment in the code.

pllim commented:
Thanks!
